diag_manager/fms_diag_axis_object.F90

```diff
@@ -0,0 +1,287 @@
+      allocate(fms_domain1d_t :: obj%axis_domain)
+      call obj%axis_domain%set(Domain=Domain)
+    else if (present(Domain2)) then
+      allocate(fms_domain2d_t :: obj%axis_domain)
+      call obj%axis_domain%set(Domain2=Domain2)
+    else if (present(DomainU)) then
+      allocate(fms_domainUG_t :: obj%axis_domain)
+      call obj%axis_domain%set(DomainU=DomainU)
+    endif
+
+    obj%tile_count = 1
+    if (present(tile_count)) obj%tile_count = tile_count
+
+    !< TO DO Check for valid domain_position
+    obj%domain_position = CENTER
+    if (present(domain_position)) obj%domain_position = domain_position
+
+    obj%length = size(axis_data)
+
+    !< TO DO Check for valid direction
+    obj%direction = 0
+    if (present(direction)) obj%direction = direction
+
+    !< TO DO Check if id is valid and with the same parameters
+    obj%edges = 0
+    if (present(edges)) obj%edges = edges
+
+    if (present(aux)) obj%aux = diag_copy_string(aux)
+    if (present(req)) obj%req = diag_copy_string(req)
+
+    obj%nsubaxis = 0
+  end subroutine diag_axis_init
+
+  !> @brief Get the length of the axis
+  !> @return axis length
+  function get_axis_length(obj) &
+  result (axis_length)
+    class(diag_axis_t), intent(inout) :: obj !< diag_axis obj
+    integer                           :: axis_length
+
+    axis_length = obj%length
+
+    !< If the axis is domain decomposed axis_length will be set to the length for the current PE:
+    if (allocated(obj%axis_domain)) axis_length = obj%axis_domain%length(obj%cart_name, obj%domain_position)
+
+  end function
+
+  !> @brief Set the subaxis of the axis obj
+  subroutine set_subaxis(obj, bounds)
+    class(diag_axis_t), INTENT(INOUT) :: obj       !< diag_axis obj
+    class(*),           INTENT(INOUT) :: bounds(:) !< bound of the subaxis
+
+    integer :: i !< For do loops
+
+    !< Check if the subaxis for this bouds already exists
+    do i = 1, obj%nsubaxis
+      if (obj%subaxis(i)%exists(bounds)) return
+    enddo
+
+    !< TO DO: everything
+    obj%nsubaxis = obj%nsubaxis + 1
+  end subroutine
+
+  !!!!!!!!!!!!!!!!!! SUB AXIS PROCEDURES !!!!!!!!!!!!!!!!!
+  !> @brief Check if a subaxis was already defined
+  !> @return Flag indicating if a subaxis is already defined
+  function check_if_subaxis_exists(obj,bounds) &
+  result(exists)
+    class(subaxis_t), INTENT(INOUT) :: obj       !< diag_axis obj
+    class(*),         INTENT(IN)    :: bounds(:) !< bounds of the subaxis
+    logical                         :: exists
+
+    !< TO DO: compare bounds
+    exists = .false.
+  end function
+
+  !> @brief Get the length of a 2D domain
+  !> @return Length of the 2D domain
+  function get_length(obj, cart_axis, domain_position) &
+  result (length)
+    class(fms_domain_t), INTENT(INOUT) :: obj       !< diag_axis obj
+    character(len=*),    INTENT(IN)    :: cart_axis !< cart_axis of the axis
+    integer,             INTENT(IN)    :: domain_position !< Domain position (CENTER, NORTH, EAST)
```

Comment:
If you are inquiring about the domain, shouldn't you already know the `cart_axis` in object you are calling it for?  I guess what I'm saying is shouldn't this function be called using the axis object instead of the domain object?  Then you wouldn't need any input arguments other than the object.  Is there a use case for this function where an axis object wouldn't/couldn't be used?